include/igraph_eulerian.h

```diff
@@ -0,0 +1,51 @@
+/* -*- mode: C -*-  */
+/*
+   IGraph library.
+   Copyright (C) 2009-2012  Gabor Csardi <csardi.gabor@gmail.com>
+   334 Harvard street, Cambridge, MA 02139 USA
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc.,  51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301 USA
+
+*/
+
+#ifndef IGRAPH_EULERIAN_H
+#define IGRAPH_EULERIAN_H
+
+#include "igraph_decls.h"
+#include "igraph_constants.h"
+#include "igraph_types.h"
+#include "igraph_vector.h"
+#include "igraph_matrix.h"
+#include "igraph_datatype.h"
+#include "igraph_iterators.h"
+#include "igraph_attributes.h"
+#include "igraph_sparsemat.h"
```

Comment:
Most of these `#include`s are not needed. Use only those which are required by the header itself. If any are required only by the `.c` file, add them only to the `.c` file, not the header.